design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,508 @@
+
+## Useful links
+
+- [Feature doc](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1266)
+
+## Issue description
+
+Many of our customers are using Azure as their main cloud provider. Therefore we want to enable Azure resources to authenticate according to Azure 
+properties and get back a Conjur access token in order to retrieve secrets from our providers. At current, customers can authenticate their Azure 
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the [feature doc](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1266), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+    annotations: 
+      provider-uri: https://sts.windows.net/TENANT_ID
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+and add the `provider-uri` annotation to the webservice (e.g `https://sts.windows.net/TENANT_ID`). 
+
+As part of our effort to improve the user experience for our authenticators, we discussed two options for defining the `provider-uri`:
+ 
+1. `provider-uri` as a variable_+
+    ```yaml
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+      - !webservice
+    
+      - !variable
+        id: provider-uri      
+    ```
+    
+1. `provider-uri` as an entry in the webservice annotations
+    ```yaml
+    - !policy
+      id: conjur/authn-azure/prod
+      body:
+        - !webservice
+          annotations: 
+            provider-uri: https://sts.windows.net/TENANT_ID
+          
+    ```     
+
+Because the `provider-uri` does not hold sensitive information, it does not need to saved in Conjur. Adding the `provider-uri` as a variable demands later that the
+admin user populate its value through the CLI. Supplying `provider-uri` as annotation entry makes it easier to set it as a key-value pair and is therefore more readable. 
+For these reasons, we have decided to add provide the `provider-uri`.
```

Comment:
I would also add that there is no increase in performance. Because the webservice itself is already retrieved  then it's a difference between retrieving an annotation from the DB and retrieving a secret from the DB.

Reply:
If I am understanding correctly, there is no increase in performance because regardless we need to retrieve info from the DB, be it an webservice or a variable?

Reply:
When retrieving an entry from DB do we also get there annotations? Or is that a separate table?